cmd/di.go

```diff
@@ -244,11 +251,24 @@ func (di *Dependencies) bootstrapNodeComponents(nodeOptions node.Options) {
 		return dialogEstablisher.EstablishDialog(providerID, contact)
 	}
 
-	promiseIssuerFactory := func(issuerID identity.Identity, dialog communication.Dialog) connection.PromiseIssuer {
-		if nodeOptions.ExperimentPromiseCheck {
-			return promise_noop.NewPromiseIssuer(issuerID, dialog, di.SignerFactory(issuerID))
+	paymentIssuerFactory := func(
+		initialState promise.State,
+		messageChan chan balance.Message,
+		dialog communication.Dialog,
+		consumer, provider identity.Identity) (connection.PaymentManager, error) {
+		// if the flag ain't set, just return a noop balance tracker
+		// it's not even a session balance tracker, but we don't really care here as long as it does nothing
+		if !nodeOptions.ExperimentPayments {
+			return payments_noop.NewSessionBalance(), nil
 		}
-		return &promise_noop.FakePromiseEngine{}
+
+		bl := balance.NewListener(messageChan)
+		ps := promise.NewSender(dialog)
+		issuer := issuers.NewLocalIssuer(di.SignerFactory(consumer))
+		tracker := promise.NewConsumerTracker(initialState, consumer, provider, issuer)
+		payments := session_payment.NewSessionPayments(messageChan, ps, tracker)
+		err := dialog.Receive(bl.GetConsumer())
+		return payments, err
```

Comment:
we could use errors.Wrap(err, "fail to receive from consumer") or something like that.